libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/accounts/UserAccountManagerTest.java

```diff
@@ -121,175 +78,126 @@ public void tearDown() throws Exception {
-        Assert.assertEquals("There should be 1 authenticated user", 1, users.size());
+        UserAccount firstUser = createTestAccountInAccountManager();
+        UserAccount secondUser = createOtherTestAccountInAccountManager();
+        userAccMgr.signoutUser(firstUser, null, false, OAuth2.LogoutReason.USER_LOGOUT);
+        Assert.assertEquals("There should be 1 authenticated user", 1, userAccMgr.getAuthenticatedUsers().size());
+        checkSameUserAccount(secondUser, userAccMgr.getCurrentUser());
     }
 
     /**
      * Removes any existing accounts.
      */
     private void cleanupAccounts() throws Exception {
-        clientManager.removeAccounts(accMgr.getAccountsByType(ClientManagerTest.TEST_ACCOUNT_TYPE));
+        for (Account acc: accMgr.getAccountsByType(ClientManagerTest.TEST_ACCOUNT_TYPE)) {
+            accMgr.removeAccountExplicitly(acc);
+        }
     }
 
     /**
      * Create a test account.
      *
-     * @return Bundle.
+     * @return UserAccount.
      */
-    private Bundle createTestAccount() {
-        return clientManager.createNewAccount(ClientManagerTest.TEST_ACCOUNT_NAME,
-        		ClientManagerTest.TEST_USERNAME, TEST_REFRESH_TOKEN,
-        		TEST_AUTH_TOKEN, TEST_INSTANCE_URL,
-        		TEST_LOGIN_URL, TEST_IDENTITY_URL,
-        		TEST_CLIENT_ID, TEST_ORG_ID,
-        		TEST_USER_ID, null, null, null,
-                null, null, null, null, null,
-                null, null, null, null, null,
-                null, null, null, false, TEST_LANGUAGE, TEST_LOCALE);
+    private UserAccount createTestAccountInAccountManager() {
+        UserAccount userAccount = UserAccountTest.createTestAccount();
+        userAccMgr.createAccount(userAccount);
+        return userAccount;
     }
 
     /**
      * Create a test account.
      *
-     * @return Bundle.
+     * @return UserAccount.
+     */
+    private UserAccount createOtherTestAccountInAccountManager() {
+        UserAccount userAccount = UserAccountTest.createOtherTestAccount();
+        userAccMgr.createAccount(userAccount);
+        return userAccount;
+    }
+
+    /**
+     * Check the user accounts are the same
+     * @param expected Expected UserAccount
+     * @param actual Actual UserAccount
      */
-    private Bundle createOtherTestAccount() {
-        return clientManager.createNewAccount(TEST_OTHER_ACCOUNT_NAME,
-        		TEST_OTHER_USERNAME, TEST_REFRESH_TOKEN,
-        		TEST_AUTH_TOKEN, TEST_INSTANCE_URL,
-        		TEST_LOGIN_URL, TEST_IDENTITY_URL,
-        		TEST_CLIENT_ID, TEST_ORG_ID_2,
-        		TEST_USER_ID_2, null, null, null,
-                null, null, null, null, null,
-                null, null, null, null, null,
-                null, null, null, false,
-                TEST_LANGUAGE, TEST_LOCALE);
+    private void checkSameUserAccount(UserAccount expected, UserAccount actual) {
+        BundleTestHelper.checkSameBundle("Not the expected user account", expected.toBundle(), actual.toBundle());
```

Comment:
We are comparing every fields (previously we were doing a Assert.equals but the UserAccount's equals method only looks at userId and orgId).